packages/common-ethereum/src/codegen/codegen-controller.ts

```diff
@@ -52,9 +52,19 @@ function validateAbi(datasources: SubqlRuntimeDatasource[], projectPath: string)
     const abiName = datasource.options.abi;
     const topicIssues: string[] = [];
     const funcIssues: string[] = [];
-    const abi = datasource.assets.get(abiName);
+
+    if (!datasource.assets) {
+      issues.push(`Databases.Asset is undefined`);
+      continue;
+    }
```

Comment:
I think line 63 can be `const abi = datasource.assets?.get(abiName);` This error isn't very useful